fix: Many fixes for enterprise IDAs

pwnage101 opened the pull request:
It seems like the various enterprise-specific devstack settings are a mix of legacy stuff combined with failed attempts to port repo-specific devstack configs into this central devstack repo. As a result, the various hostnames and paths are completely broken. This hasn't impacted us because we normally don't use enterprise inside of devstack, but this is will hopefully be our future.

clintonb commented on provision.sh:
```diff
@@ -38,13 +39,14 @@ echo -e "${GREEN}Creating databases and users...${NC}"
-./provision-ecommerce.sh
-./provision-discovery.sh
-./provision-credentials.sh
+cat <<CMDS | xargs --max-procs=0 --max-lines=1 -I {} bash -c {}
```
s/--max-lines=/-L

Use -P instead of max-procs. Unfortunately, the macOS version does not support maxing out by setting the value to 0.

pwnage101 committed:
fix: Many fixes for enterprise IDAs

It seems like the various enterprise-specific devstack settings are a
mix of legacy stuff combined with failed attempts to port repo-specific
devstack configs into this central devstack repo. As a result, the
various hostnames and paths are completely broken. This hasn't impacted
us because we normally don't use enterprise inside of devstack, but this
is will hopefully be our future.

pwnage101 commented on Makefile:
```diff
@@ -406,7 +406,7 @@ dev.shell.insights:
 
 dev.shell.%: ## Run a shell on the specified service's container.
-	docker compose exec $* /bin/bash
+	docker compose exec -u 0 $* /bin/bash
```
We need to be logging developers into the containers as root, NOT as the unprivileged app user.  Without root, you're basically helpless as you can't install custom requirements and other basic things like run management commands from the shell.

We've been doing this in the various Makefiles in the enterprise-* repos since forever.

pwnage101 commented on docker-compose-host.yml:
```diff
@@ -24,10 +24,12 @@ services:
       - ${DEVSTACK_WORKSPACE}/enterprise-access:/edx/app/enterprise-access/
       - ${DEVSTACK_WORKSPACE}/src:/edx/src
-      - ${PWD}/py_configuration_files/enterprise_access.py:/edx/app/enterprise-access/license_manager/settings/devstack.py
+      - ${PWD}/py_configuration_files/enterprise_access.py:/edx/app/enterprise-access/enterprise_access/settings/devstack.py
```
💩
